src/components/main/Scenario.js

```diff
@@ -35,54 +39,122 @@ class Scenario extends React.Component {
+                }
+
+            }
+
+	}
 	
+	setInfo = (name,val) =>
+	{
+		let info=this.state.info;
+		if(name==='unit')
+		{
+			info.unit=val;
+		}else if(name==='HPPD')
+		{
+			info.HPPD=val;
+		}else if(name==='census')
+		{
+			info.census=val;
+		}else if(name==='bedUnit')
+		{
+			info.bedUnit=val;
+		}
+		this.setState({info:info});
+		this.calculation(info,this.state.num);
+	}
 	setInfoStaffNum = (num) =>
 	{
 		debugger;
 		this.setState({"num":num});
-		// this.calculation(this.state.info,num);
+		this.calculation(this.state.info,num);
 	}
+
 	
+	random = (Min,Max) =>{
+		var Range = Max - Min;   
+		var Rand = Math.random();   
+		return (Min + Math.round(Rand * Range));
+	}
+	handleRandom = () =>
+	{
+		const num1 = this.random(8,24);
+		var a=document.getElementById("HPPD");
+        a.value=num1;
+		this.setInfo("HPPD",num1);	
+		
+		const num2 = this.random(1,100);
+		a=document.getElementById("census");
+        a.value=num2;
+	    this.setInfo("census",num2)	;
+		const num3 = this.random(10,1000);
+		a=document.getElementById("bedUnit");
+        a.value=num3;
+		this.setInfo("bedUnit",num3)	;
+    }
+
     render() {
         return (
-
             <div className="App">
 
-                    <Card id="scenario">
-                        <Card.Header>Fill out the scenario fields based off of the in-class example.</Card.Header>
-						<Card.Body>
-							<Form>
-								<InputGroup size="sm" className="mb-3">
-									<p>The hospital unit is </p>
-									<input type='text' name='unit' data-testid="unit-id" onChange={this.changeHandler.bind(this)}/>
-									<p> and the HPPD is </p>
-									<input type='text' name='hppd' data-testid="hppd-id" onChange={this.changeHandler.bind(this)}/>
-									<p>. You have </p>
-									<input type='text' name='numbeds' data-testid="numbeds-id" onChange={this.changeHandler.bind(this)}/>
-									<p> number of beds in your unit and your census is </p>
-									<input type='text' name='census' data-testid="census-id" value={this.state.census} onChange={this.changeHandler.bind(this)}/>
-									<p>% full. Based off of this scenario, allocate your staffing resources.</p>
-								</InputGroup>
-							</Form>
-						</Card.Body>
-					</Card>
-
-            		<StaffAdd staffs={this.state.staffs}  results={this.state.results} staffNum={this.state.staffNum} setInfoStaffNum={this.setInfoStaffNum}/>
-            </div>
+                <Card id="scenario">
+                    <Card.Header>Fill out the scenario fields based off of the in-class example.</Card.Header>
+                    <Card.Body>
+                        <Form>
+                            <InputGroup size="sm" className="mb-3">
+                                <p>The hospital unit is</p>
+                                <input type='text' name='unit' data-testid="unit-id" onChange={this.changeHandler.bind(this)} />
+                                <p>and the HPPD is</p>
```

Comment:
The types were removed from the input fields. Without the types on the input field it breaks the scenario.test.js suite. This is because in the test it make as call to `.toHaveAttribute("type", "text");`. The type text is what is needed for the tests  as it will allow for text and numbers. There is a type = numbers but it changed the layout and since we were doing input checking the type = text should be fine in this case.

Reply:
fix